src/gstspotifysrc.c

```diff
@@ -220,23 +194,32 @@ gst_spotify_src_class_init (GstSpotifySrcClass * klass)
   basesrc_class->unlock_stop = gst_spotify_src_unlock_stop;
   basesrc_class->do_seek = gst_spotify_src_do_seek;
   basesrc_class->is_seekable = gst_spotify_src_is_seekable;
-  basesrc_class->check_get_range = gst_spotify_src_check_get_range;
   basesrc_class->get_size = gst_spotify_src_do_get_size;
   basesrc_class->query = gst_spotify_src_query;
 
+  gst_element_class_add_pad_template (element_class,
+    gst_static_pad_template_get (&gst_spotify_src_template));
+
+  gst_element_class_set_metadata (
+    element_class,
+    "Spotify audio source",
+    "Source/Audio/Network/Protocol",
+    "Feed Spotify hosted music to a pipeline",
+    "Liam Wickins <liam9534@gmail.com>");
+
+  GST_DEBUG_CATEGORY_INIT (spotify_src_debug, "spotify", 0, "spotifysrc element");
+
   g_type_class_add_private (klass, sizeof (GstSpotifySrcPrivate));
 }
 
 static void
-gst_spotify_src_init (GstSpotifySrc * spotifysrc, GstSpotifySrcClass * klass)
+gst_spotify_src_init (GstSpotifySrc * spotifysrc)
 {
   GstSpotifySrcPrivate *priv;
 
-  priv = spotifysrc->priv = G_TYPE_INSTANCE_GET_PRIVATE (spotifysrc, GST_TYPE_SPOTIFY_SRC,
-      GstSpotifySrcPrivate);
+  priv = spotifysrc->priv = G_TYPE_INSTANCE_GET_PRIVATE (
+        spotifysrc, GST_TYPE_SPOTIFY_SRC, GstSpotifySrcPrivate);
 
-  priv->mutex = g_mutex_new ();
```

Comment:
https://developer.gnome.org/glib/stable/glib-Deprecated-Thread-APIs.html#g-mutex-new

g_mutex_free has been deprecated since version 2.32 and should not be used in newly-written code.
GMutex can now be statically allocated, or embedded in structures and initialised with g_mutex_init().